compiler/rustc_borrowck/src/consumers.rs

```diff
@@ -22,8 +22,6 @@ pub use super::{
 ///     because they are evaluated during typechecking. The panic can be avoided
 ///     by overriding the `mir_borrowck` query. You can find a complete example
 ///     that shows how to do this at `src/test/run-make/obtain-borrowck/`.
-/// *   This function will also panic if computation of Polonius facts
-///     (`-Zpolonius` flag) is not enabled.
```

Comment:
I think the purpose of this is to ensure that we don't accidentally invoke this query on the "happy-path". It seems fine to me to remove this constraint, since we have `perf` to check for those kind of regressions, but could you add a note to the first paragraph of this comment indicating that this should never be necessary for a typical compilation session?